### ISSUE TITLE
Improvements to the Video Player

### ISSUE THREAD

**Issue by De-M-oN** [rejetto/hfs#835]
I'd like to suggest a few improvements to the videoplayer:

- Arrow keys Left & Right should seek only 5 or 10 seconds, not a whole minute
- Left & Right Arrowkeys should be more reliable. Sometimes if not the right thing is in focus it would switch to next video instead. Up and down arrow just changes which element is in focus while the help box says up and down is for seeking I guess? (its a bit weird translated to german), but that never worked. It always just changes which element is in focus.
- I wish Spacebar would be a pause button and not just left click.
- Hotkey for volume up and down
- maybe its possible to get a link that the player opens directly instead of having to give the filelist link and telling them to click on the videofile with windows key pressed down?

Not sure if the following are in any way possible outside of a video platform but cool would also be:

- minithumbs on the seekbar
- chapters on the seekbar
- remembering the playback position so that you can continue watching later from the position you stopped.
- rightclick into the video and copy a URL with playback position would be then the next step from last point of previous list.

**Comment by rejetto** [comment]
> Arrow keys Left & Right should seek only 5 or 10 seconds, not a whole minute

the videoplayer is (currently) the one of the browser, HFS does basically *nothing*.
I don't exclude in the future to customize it somehow. 

> Left & Right Arrowkeys should be more reliable.

HFS' file-show (which is not only for videos) uses the arrows to navigate between files.
When the player is focused, it surely uses the same keys for seeking. I'm not sure what's the right thing to do here. If you go full-screen, the problem should go away.

> Up and down arrow just changes which element is in focus

that's not HFS doing it. HFS uses up/down only for images at the moment. Maybe i can try to use up/down for 10-second seeking, or for the volume.

> its a bit weird translated to german

next version has a option to force english inside the "options" dialog.

> I wish Spacebar would be a pause button and not just left click.

in chrome, when the video player is focused, spacebar does pause.

> maybe its possible to get a link that the player opens directly

but, what happens if you give a direct link to a video file?
in my tests, the browser plays the video directly.

> remembering the playback position so that you can continue watching later from the position you stopped.

this should be feasible

**Comment by De-M-oN** [comment]
> in chrome, when the video player is focused, spacebar does pause.

Yes but then a white border is around the player window even at fullscreen - (probably the element being selected highlighting)

> but, what happens if you give a direct link to a video file?

Then it uses the most basic player you can imagine by the browser and not your own implementation (file-show)
Any my MKV files it would open download file dialog rather than playback it.

> this should be feasible

Thats surprising to hear, nice if it is.

Another point I have is:

Can you make it so, that when you hover the mouse over the seekbar, that it shows the time? Because without it you're blindly clicking around to find e.g. minute 24.

**Comment by rejetto** [comment]
> Yes but then a white border is around the player window even at fullscreen - (probably the element being selected highlighting)

the focus can be hidden using style, if you don't like it

> Then it uses the most basic player you can imagine by the browser and not your own implementation (file-show)

but file-show is doing very little for playing videos, at the moment. I doubt you are missing any feature of it, if you are playing just 1 file.

> Can you make it so, that when you hover the mouse over the seekbar, that it shows the time? 

i doubt that.

HFS is not designed as media-player, so advanced "media" features are not part of the main goals. But plugins can do much.
If one has advanced necessities, should probably consider a media server. I'm using jellyfin.

**Comment by De-M-oN** [comment]
> but file-show is doing very little for playing videos, at the moment. I doubt you are missing any feature of it, if you are playing just 1 file.

But your file-show playbacks MKV files
While otherwise it would offer to download the file

> I'm using jellyfin

hmm will look into it what it is

**Comment by rejetto** [comment]
> While otherwise it would offer to download the file

what browser? chrome seems to wider support for mkv.
are you using the "unsupported-videos" plugin?

**Comment by De-M-oN** [comment]
Google Chrome.

file-show: plays it
without your file-show: Download file dialog appears^^

> are you using the "unsupported-videos" plugin?

no.

**Comment by rejetto** [comment]
ok, but you should probably try that plugin if you are interested in having a wider support. with it the unsupported files only gets "play" but not seeking.

**Comment by hugo529100** [comment]
waterfox seem better than chrome
